main/src/cgeo/geocaching/settings/Settings.java

```diff
@@ -220,12 +223,58 @@ private static void migrateSettings() {
             e.putString(getKey(R.string.pref_defaultNavigationTool2), String.valueOf(dnt2));
 
             // defaults for gpx directories
-            e.putString(getKey(R.string.pref_gpxImportDir), getGpxImportDir());
-            e.putString(getKey(R.string.pref_gpxExportDir), getGpxExportDir());
+            e.putString(getKey(R.string.pref_gpxImportDir), LocalStorage.getDefaultGpxDirectory().getPath());
+            e.putString(getKey(R.string.pref_gpxExportDir), LocalStorage.getDefaultGpxDirectory().getPath());
 
             e.putInt(getKey(R.string.pref_settingsversion), 2); // mark migrated
             e.apply();
         }
+
+        if (currentVersion < 3) {
+            final Editor e = sharedPrefs.edit();
+
+            // move field-notes
+            FileUtils.move(LocalStorage.getLegacyFieldNotesDirectory(), LocalStorage.getFieldNotesDirectory());
+
+            // move gpx files
+            if (getGpxExportDir().equals(LocalStorage.getLegacyGpxDirectory().getPath())) {
+                e.putString(getKey(R.string.pref_gpxExportDir), LocalStorage.getDefaultGpxDirectory().getPath());
+            }
+            if (getGpxImportDir().equals(LocalStorage.getLegacyGpxDirectory().getPath())) {
+                e.putString(getKey(R.string.pref_gpxImportDir), LocalStorage.getDefaultGpxDirectory().getPath());
+            }
+            FileUtils.move(LocalStorage.getLegacyGpxDirectory(), LocalStorage.getDefaultGpxDirectory());
+
+            // local spoilers
+            FileUtils.move(LocalStorage.getLegacyLocalSpoilersDirectory(), LocalStorage.getLocalSpoilersDirectory());
+
+            // move db files
+            FileUtils.moveTo(new File(LocalStorage.getLegacyExternalCgeoDirectory(), DataStore.DB_FILE_NAME), LocalStorage.getExternalDbDirectory());
+            FileUtils.moveTo(new File(LocalStorage.getLegacyExternalCgeoDirectory(), DataStore.DB_FILE_NAME + DataStore.DB_FILE_CORRUPTED_EXTENSION), LocalStorage.getBackupDirectory());
+            FileUtils.moveTo(new File(LocalStorage.getLegacyExternalCgeoDirectory(), DataStore.DB_FILE_NAME_BACKUP), LocalStorage.getBackupDirectory());
+            FileUtils.moveTo(new File(LocalStorage.getLegacyExternalCgeoDirectory(), DataStore.DB_FILE_NAME_BACKUP + DataStore.DB_FILE_CORRUPTED_EXTENSION), LocalStorage.getBackupDirectory());
+
+            // move geocache data files
+            final FileFilter geocacheDirectories = new FileFilter() {
+                @Override
+                public boolean accept(final File pathname) {
+                    final String name = pathname.getName();
+                    return pathname.isDirectory() &&
+                            (HtmlImage.SHARED.equals(name) || LocalStorage.GEOCACHE_FILE_PATTERN.matcher(name).find());
+                }
+            };
+            final File[] list = LocalStorage.getLegacyExternalCgeoDirectory().listFiles(geocacheDirectories);
```

Comment:
On the other hand, if `getLegacyExternalCgeoDirectory()` is supposed to be a directory, then we might want to crash if it isn't :)